src/dispatch.rs

```diff
@@ -100,17 +104,26 @@ impl Dispatcher {
         if self.connections.contains_key(&req.db_name[..]) {
             return Ok("".into());
         }
-        match IdbStore::new(&req.db_name[..]).await {
-            Err(e) => {
-                return Err(format!("Failed to open \"{}\": {}", req.db_name, e));
-            }
-            Ok(v) => {
-                if let Some(kv) = v {
-                    self.connections
-                        .insert(req.db_name.clone(), Box::new(dag::Store::new(Box::new(kv))));
-                }
+        let idb_store = match IdbStore::new(&req.db_name[..]).await {
+            Err(e) => return Err(format!("Failed to open \"{}\": {}", req.db_name, e)),
+            Ok(None) => return Err(format!("Failed to open \"{}\"", req.db_name)),
+            Ok(Some(idb_store)) => {
+                let _client_id = match init_client_id(&idb_store).await {
+                    Ok(v) => v,
+                    Err(why) => {
+                        return Err(format!("Failed to read client ID: {:?}", why));
+                    }
+                };
+                idb_store
             }
-        }
+        };
+
+        self.connections.insert(
+            req.db_name.clone(),
+            Box::new(dag::Store::new(Box::new(idb_store))),
+        );
+
+        // TODO(arv): If we get None in IdbStore::new that should be an error.
```

Comment:
I think you handled this, right?

Reply:
I want to change IdbStore::new to return an Error but I'm OK dealing with it here.